### DISCUSSION TITLE
Making you GUI from grund up

### DISCUSSION THREAD

**Discussion by logos-maker** [free-audio/clap#135] [Q&A] [answered]
For your GUI, is it possible to only rely on calls from the host for your GUI or do you have to fork a function call from the host to get a process to run your GUI? And what function is it the n that you should fork. Or is this question totally misplaced? Is there something that is needed to be understood about  thread-pool? What is the function of thread pool, and how do use it?

**Comment by baconpaul** [comment, marked as answer]
The thread-pool is for audio processing, not for GUI processing. It is documented in the thread pool extension header with an example.

On mac and windows, there is the standard 'main thread' concept. If your plugin implements the gui extension documented here https://github.com/free-audio/clap/blob/main/include/clap/ext/gui.h you will get a callback on the main thread to reparent a window

On linux you will almost definitely need to implement the timer and perhaps posix-fd extensions to get callbacks from the host for idle timers or for file descriptors.

Hope that helps!
